pytamaro/graphic.py

```diff
@@ -242,15 +274,78 @@ class Rotate(Graphic):
     """
     Represents the rotation of a graphic by a certain angle clockwise.
     """
+
     def __init__(self, graphic: Graphic, deg: float):
         self.graphic = graphic
         self.rot_matrix = Matrix.RotateDeg(deg, self.graphic.pin_position)
         self.path = Path()
         self.graphic.path.transform(self.rot_matrix, self.path)  # updates self.path
         self.pin_position = graphic.pin_position
+        # For graphic tree comparison
+        self.degree = - deg
 
     def draw(self, canvas: Canvas):
         canvas.save()
         canvas.concat(self.rot_matrix)
         self.graphic.draw(canvas)
         canvas.restore()
+
+
+class Beside(Graphic):
+    """
+    Represent the composition of two graphics that beside with each other,
+    the center of the two graphics are on the same horizontal level.
+    """
+
+    def __init__(self, left_graphic: Graphic, right_graphic: Graphic):
+        self.left_graphic = left_graphic
+        self.right_graphic = right_graphic
+
+        b_g = Pin(Compose(Pin(left_graphic, center_right), Pin(right_graphic, center_left)), center)
+
+        self.path = Path(b_g.path)
+        self.set_pin_position(b_g.pin_position.x(), b_g.pin_position.y())
+
+    def draw(self, canvas: Canvas):
+        b_g = Pin(Compose(Pin(self.left_graphic, center_right), Pin(self.right_graphic, center_left)), center)
+        b_g.draw(canvas)
+
+
+class Above(Graphic):
+    """
+    Represent the composition of two graphics that one above other,
+    the center of the two graphics are on the same vertical level.
+    """
+
+    def __init__(self, top_graphic: Graphic, bottom_graphic: Graphic):
+        self.top_graphic = top_graphic
+        self.bottom_graphic = bottom_graphic
+
+        a_g = Pin(Compose(Pin(top_graphic, bottom_center), Pin(bottom_graphic, top_center)), center)
+
+        self.path = Path(a_g.path)
+        self.set_pin_position(a_g.pin_position.x(), a_g.pin_position.y())
+
+    def draw(self, canvas: Canvas):
+        a_b = Pin(Compose(Pin(self.top_graphic, bottom_center), Pin(self.bottom_graphic, top_center)), center)
+        a_b.draw(canvas)
+
+
+class Overlay(Graphic):
+    """
+    Represent the composition of two graphics that one overlay other,
+    the center of the two graphics are at the same position
+    """
+
+    def __init__(self, front_graphic: Graphic, back_graphic: Graphic):
+        self.front_graphic = front_graphic
+        self.back_graphic = back_graphic
+
+        o_g = Pin(Compose(Pin(front_graphic, center), Pin(back_graphic, center)), center)
```

Comment:
If I'm not missing something, I think one should be able to factor out this common logic among the three classes (e.g., in an intermediate class in the hierarchy named something like `SimpleCompose` which extends `Graphic` and from which `Above` `Beside` and `Overlay` are further derived).
They could share the same `draw` method, which would make use of the shared `composed_graphic` attribute.